packages/multi-test/src/contracts.rs

```diff
@@ -354,13 +354,23 @@ where
         info: MessageInfo,
         msg: Vec<u8>,
     ) -> AnyResult<Response<C>> {
-        let msg = from_slice(&msg)?;
-        (self.instantiate_fn)(deps, env, info, msg).map_err(|err| anyhow!(err))
+        let msg: T2 = from_slice(&msg)?;
+        (self.instantiate_fn)(deps, env, info, msg.clone())
```

Comment:
Just a thought, we could remove the `Clone` bound if you did the following, right?

```rust
let ctx = format!("Contract returned an error on instantiate msg:\n{:?}", msg);
(self.instantiate_fn)(deps, env, info, msg).map_err(anyhow::Error::from).context(ctx)
```

Nothing really important, just like reducing required bounds and removing clones when possible.

Reply:
Huh, yeah, that looks much better. Will do :+1: